src/core/net/ip6.cpp

```diff
@@ -912,10 +912,10 @@ Error Ip6::HandleExtensionHeaders(Message &    aMessage,
             break;
 
         case kProtoFragment:
-            // Always forward IPv6 fragments to the Host.
+#if !OPENTHREAD_CONFIG_IP6_FRAGMENTATION_ENABLE
```

Comment:
This seems good and reasonable to me. But I noticed that there is a comment `// Always forward IPv6 frgrament to the Host`. Checking the history I think it was added from 
- #4940

which was authord by @LuDuda.

@LuDuda can you please review this PR and check if you are ok with this change?

Reply:
Looks good for me, in our use case, we have always `OPENTHREAD_CONFIG_IP6_FRAGMENTATION_ENABLE` disable on NCP.